### ISSUE TITLE
RandomCutMixV2: cut placement is one uniform draw shared by every row and mix, even with same_on_batch=False (regression from #1459)

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4712]
## Describe the bug

`CutmixGenerator.forward` draws **one** uniform number per axis for the cut start position and shares it across every row and every mix of the batch, even with `same_on_batch=False`. Only the cut *sizes* are drawn per row. The placement of all `B * num_mix` cuts is therefore perfectly correlated: every cut sits at the same relative position `u * (side - cut_side - 1)`.

```python
_gen_shape = (1,)
if same_on_batch:
    _gen_shape = (cut_height.size(0),)
    ...
x_start = _adapted_rsampling(_gen_shape, self.rand_sampler, same_on_batch) * (width - cut_width - 1)
```

`rand_sampler` is a scalar `Uniform(0, 1)`, so `_gen_shape = (1,)` yields a single draw that is broadcast against the per-row `cut_width`. With `same_on_batch=True`, `_adapted_rsampling` also repeats one draw, so both modes use one placement draw; they differ only in the sizes.

## Reproduction

`main` @ `c90c05f66`, torch 2.14.0, CPU, macOS arm64:

```python
import torch, kornia.augmentation as K
from kornia.geometry.bbox import infer_bbox_shape

H = W = 100000  # large, so floor() is negligible
torch.manual_seed(0)
aug = K.RandomCutMixV2(p=1.0, num_mix=2, same_on_batch=False, use_correct_lambda=True)
c = aug.forward_parameters(torch.Size([4, 1, H, W]))["crop_src"].double().view(-1, 4, 2)
h, w = infer_bbox_shape(c)
print(w.tolist())
print((c[:, 0, 0] / (W - w - 1)).tolist())
print((c[:, 0, 1] / (H - h - 1)).tolist())
```

```
widths          [78942, 95768, 87808, 85240, 85751, 51559, 76970, 84918]      # per-row, as expected
x0 / (W-w-1)    [0.03400, 0.03380, 0.03396, 0.03401, 0.03397, 0.03402, 0.03400, 0.03402]   # one draw
y0 / (H-h-1)    [0.94420, 0.94422, 0.94422, 0.94424, 0.94420, 0.94424, 0.94424, 0.94423]   # one draw
```

With a fixed cut size (`cut_size=(0.5, 0.5)`) the consequence is that `crop_src` is identical across all 16 cuts of a `B=8, num_mix=2` batch in 200 of 200 seeds with `same_on_batch=False`, i.e. `same_on_batch` has no observable effect on the geometry there.

## Expected behavior

With `same_on_batch=False`, each row and mix draws its own start position, as it already draws its own size.

## Provenance: a regression, not the original design

The original implementation (#609, `fe6c54c95`) called `_adapted_uniform((1,), zeros_like(cut_width), width - cut_width, same_on_batch)` with **batch-shaped** `low`/`high`, so a sample shape of `(1,)` produced `(1, B * num_mix)` independent draws. The random-generator refactor (#1459, `fb7565a2f`) replaced that with a scalar `Uniform(0, 1)` sampler times a scale and kept `_gen_shape = (1,)`, which collapsed the placement to one draw.

## Suggested fix

Sample `(batch_size * num_mix,)` start positions when `same_on_batch=False`. This changes the RNG stream of `RandomCutMixV2` for a fixed seed, so it wants a `changelog.d` `fixed` fragment, and the doctest on `RandomCutMixV2` plus any seeded expected values need regenerating. A pin should use an input where sizes are fixed (`cut_size=(0.5, 0.5)`) so that only the placement can differ, and assert that not all `crop_src` rows are equal; the `same_on_batch=True` pin should use the default `cut_size` so that sizes are the discriminating part.

Found while reviewing #4711.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
